test/test_migrate.py

```diff
@@ -15,24 +14,13 @@ def run_qtile_migrate(config):
     subprocess.check_call(argv)
 
 
-def hash_file(p):
-    with open(p) as f:
-        h = hashlib.sha256()
-        h.update(f.read().encode('utf-8'))
-        return h.digest()
-
-
 def test_migrate_default_config_noop():
     with tempfile.TemporaryDirectory() as temp:
         default_config = os.path.join(os.path.dirname(__file__), '..', 'libqtile', 'resources', 'default_config.py')
         config_path = os.path.join(temp, "config.py")
         shutil.copyfile(default_config, config_path)
         run_qtile_migrate(config_path)
-
-        original = hash_file(config_path)
```

Comment:
I wonder if we should hash it before we migrate it, so we can make sure it's still a no-op?

Reply:
[filecmp](https://docs.python.org/3/library/filecmp.html#filecmp.cmp) compares the contents of the files, so I'm not sure what hashing would also provide. Unless I misunderstand what you mean?

Reply:
Oh, I see what you mean actually, OK, ignore the above

Reply:
I just added it back, but reading through it on here again I think it's actually redundant, but perhaps I can make the test check for the existence of the backup file. If it's there, then `filecmp.cmp` returned `False` -> the files differ.

Reply:
Alright try this out for size

Reply:
LGTM, thanks!